src/serac/infrastructure/accelerator.hpp

```diff
@@ -75,6 +77,87 @@ enum class ExecutionSpace
  */
 constexpr ExecutionSpace default_execution_space = ExecutionSpace::CPU;
 
+namespace detail {
+
+/**
+ * @brief Trait for "translating" between serac::ExecutionSpace and axom::MemorySpace
+ */
+template <ExecutionSpace space>
+struct execution_to_memory {
+  /// @brief The corresponding axom::MemorySpace
+  static constexpr axom::MemorySpace value = axom::MemorySpace::Dynamic;
+};
+
+#ifdef SERAC_USE_UMPIRE
+/// @overload
+template <>
+struct execution_to_memory<ExecutionSpace::CPU> {
+  static constexpr axom::MemorySpace value = axom::MemorySpace::Host;
+};
+
+/// @overload
+template <>
+struct execution_to_memory<ExecutionSpace::GPU> {
+  static constexpr axom::MemorySpace value = axom::MemorySpace::Device;
+};
+
+/// @overload
+template <>
+struct execution_to_memory<ExecutionSpace::Dynamic> {
+  static constexpr axom::MemorySpace value = axom::MemorySpace::Unified;
+};
```

Comment:
What is the interpretation of "Dynamic" and "Unified" here? Does this correspond to nvidia's notion of a unified memory model, or is it something else?

Reply:
Yes - previously ExecutionSpace::Dynamic corresponded to a cudaMallocManaged in `serac::Array`, so the new correspondence maintains that same functionality.